### PR TITLE
Feature/sdk status

### PR DESCRIPTION
**Pull request by ralf601**

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -1,123 +1,338 @@
...
+        if (sdkInitialized) {
+            const sdkStatus = await RNSentiance.getSdkStatus();
+            this.onSdkStatusUpdate(sdkStatus);
+            this.updateUserIdAndSdkVersion();
+            this.listenToUserActivity();
+        }
+    }
+
+    async onSdkStatusUpdate(sdkStatus) {
+      console.log("onSdkStatusUpdate");
```
Comment (robonaut) [inline App.js]: Please remove console.log (performance issues)

File: App.js [hunk 2]
```diff
@@ -1,123 +1,338 @@
...
+
+    onUserActivityUpdate(userActivity) {
+        var {
+            type,
+            tripInfo,
+            stationaryInfo
+        } = userActivity;
+
+        if (type === "USER_ACTIVITY_TYPE_STATIONARY") {
+            var {
```
Comment (robonaut) [inline App.js]: Prefer `const` over `var` if you can

File: App.js [hunk 3]
```diff
@@ -1,123 +1,338 @@
...
+            });
+        } else if (type === "USER_ACTIVITY_TYPE_UNKNOWN") {
+            this.setState({
+                userActivity: "Unknown"
+            });
+        }
+    }
+
+    async updateUserIdAndSdkVersion() {
+        await RNSentiance.getUserId()
```
Comment (robonaut) [inline App.js]: You can just write

`
try {
  const userId = await RNSentiance.getUserId();
} catch (err) {
  ...
}
`

File: App.js [hunk 4]
```diff
@@ -1,123 +1,338 @@
...
-    getSDKPropertiesFromBridge(setSDKProperties);
-  }, []);
-  return (
-    <Fragment>
-      <StatusBar barStyle="dark-content" />
-      <SafeAreaView>
-        <ScrollView
-          contentInsetAdjustmentBehavior="automatic"
-          style={styles.scrollView}
+    async isSdkInitiazed() {
```
Comment (robonaut) [inline App.js]: typo?

File: App.js [hunk 5]
```diff
@@ -1,123 +1,219 @@
...
+import styles from "./styles";
 
-import {
-  Header,
-  LearnMoreLinks,
-  Colors,
-  DebugInstructions,
-  ReloadInstructions
-} from "react-native/Libraries/NewAppScreen";
+rnSentianceEmitter = new NativeEventEmitter(RNSentiance);
```
Comment (jiricrispeyn) [inline App.js]: Should be `const rnSentianceEmitter = ...`

File: App.js [hunk 6]
```diff
@@ -1,123 +1,219 @@
...
+    } = sdkStatus;
+
+    const diskQuota = await RNSentiance.getDiskQuotaLimit();
+    const diskQuotaUsed = await RNSentiance.getDiskQuotaUsage();
+    const mobileQuota = await RNSentiance.getMobileQuotaLimit();
+    const mobileQuotaUsed = await RNSentiance.getMobileQuotaUsage();
+    const wifiQuota = await RNSentiance.getWiFiQuotaLimit();
+    const wifiQuotaUsed = await RNSentiance.getWiFiQuotaUsage();
+
+    data = [
```
Comment (jiricrispeyn) [inline App.js]: Why not just export an object instead of manually creating key/value pairs?

File: App.js (hunk 5 above)
Comment (robonaut) [reply App.js]: Damn, I overlooked this one. Thanks!

File: App.js (hunk 6 above)
Comment (robonaut) [reply App.js]: While ES6 / ES2015 adds property order, it does not require for-in or Object.keys to follow that order, due to legacy compatibility concerns. However we could do it without any issues. Think array is a bit more natural here though.

**Pull request merged**